File: SDKs/zuora.com/2021-08-20/go/pkg/models/shared/timeslicedtcbnetmetrics.go
package shared

import (
	"time"
)

type TimeSlicedTcbNetMetricsGeneratedReasonEnum string

const (
	TimeSlicedTcbNetMetricsGeneratedReasonEnumIncreaseQuantity TimeSlicedTcbNetMetricsGeneratedReasonEnum = "IncreaseQuantity"
	TimeSlicedTcbNetMetricsGeneratedReasonEnumDecreaseQuantity TimeSlicedTcbNetMetricsGeneratedReasonEnum = "DecreaseQuantity"
	TimeSlicedTcbNetMetricsGeneratedReasonEnumChangePrice      TimeSlicedTcbNetMetricsGeneratedReasonEnum = "ChangePrice"
	TimeSlicedTcbNetMetricsGeneratedReasonEnumExtension        TimeSlicedTcbNetMetricsGeneratedReasonEnum = "Extension"
	TimeSlicedTcbNetMetricsGeneratedReasonEnumContraction      TimeSlicedTcbNetMetricsGeneratedReasonEnum = "Contraction"
)

type TimeSlicedTcbNetMetricsTypeEnum string

const (
	TimeSlicedTcbNetMetricsTypeEnumRegular  TimeSlicedTcbNetMetricsTypeEnum = "Regular"
	TimeSlicedTcbNetMetricsTypeEnumDiscount TimeSlicedTcbNetMetricsTypeEnum = "Discount"
)

type TimeSlicedTcbNetMetrics struct {
	Amount               *float64                                    `json:"amount,omitempty"`
	DiscountChargeNumber *string                                     `json:"discountChargeNumber,omitempty"`
	EndDate              *time.Time                                  `json:"endDate,omitempty"`
	GeneratedReason      *TimeSlicedTcbNetMetricsGeneratedReasonEnum `json:"generatedReason,omitempty"`
	InvoiceOwner         *string                                     `json:"invoiceOwner,omitempty"`
	OrderItemID          *string                                     `json:"orderItemId,omitempty"`
	StartDate            *time.Time                                  `json:"startDate,omitempty"`
	SubscriptionOwner    *string                                     `json:"subscriptionOwner,omitempty"`
	Tax                  *float64                                    `json:"tax,omitempty"`
	TermNumber           *float64                                    `json:"termNumber,omitempty"`
	Type                 *TimeSlicedTcbNetMetricsTypeEnum            `json:"type,omitempty"`
}

File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/timeslicedtcbnetmetrics.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum TimeSlicedTcbNetMetricsGeneratedReasonEnum {
    IncreaseQuantity = "IncreaseQuantity",
    DecreaseQuantity = "DecreaseQuantity",
    ChangePrice = "ChangePrice",
    Extension = "Extension",
    Contraction = "Contraction"
}

export enum TimeSlicedTcbNetMetricsTypeEnum {
    Regular = "Regular",
    Discount = "Discount"
}


export class TimeSlicedTcbNetMetrics extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=amount" })
  amount?: number;

  @SpeakeasyMetadata({ data: "json, name=discountChargeNumber" })
  discountChargeNumber?: string;

  @SpeakeasyMetadata({ data: "json, name=endDate" })
  endDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=generatedReason" })
  generatedReason?: TimeSlicedTcbNetMetricsGeneratedReasonEnum;

  @SpeakeasyMetadata({ data: "json, name=invoiceOwner" })
  invoiceOwner?: string;

  @SpeakeasyMetadata({ data: "json, name=orderItemId" })
  orderItemId?: string;

  @SpeakeasyMetadata({ data: "json, name=startDate" })
  startDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=subscriptionOwner" })
  subscriptionOwner?: string;

  @SpeakeasyMetadata({ data: "json, name=tax" })
  tax?: number;

  @SpeakeasyMetadata({ data: "json, name=termNumber" })
  termNumber?: number;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: TimeSlicedTcbNetMetricsTypeEnum;
}
